### ISSUE TITLE
Feature request: time of last touch event

### ISSUE THREAD

**Issue by Radushka** [daniel-2k/ioBroker.nanoleaf-lightpanels#20]
First off, thank you for this adapter!

Is it possible to add to the gesture and PanelID a field with the latest touch event time? Trying to build an automation, but I'm not sure how to detect if the same panel is pressed the same way, i.e. ID is the same and the gesture is the same.

Thank you!

**Comment by daniel-2k** [comment]
This is already possible with the standard information which every state has in ioBroker.
If you have a trigger to the gesture or panelID state you can get the last change (this is the timestamp there the state changed last time) and the timestamp which indicated the last update of the state even when it's the same value.
This you can use. In addtion you can get also the privious value, prvious last change and provious timestamp.
With this you have all necessary information to detect a gesture if it is the same.

You can simply use a trigger which only triggers on value change instead of every update.

**Comment by Radushka** [comment]
I am just finding my way around iobroker. Thanks for the clues which helped me read the 'ts' and 'lc' properties of the gesture and PanelID. I think I have everything I need now 👍

**Comment by Radushka** [comment]
Hi @daniel-2k, still struggling with this one. I got it all set up with the Home Assistant pulling the ts values via a REST request. It all works fine, except for the delay I'm trying to tackle. So I tried the MQTT adapter route to send state updates by subscribing to the Nanoleaf adapter and publishing full object states. To my disappointment, the MQTT adapter "senses" the change in the state only if the 'val' of the state has changed. So here I am again kindly requesting if you would consider adding a couple of new attributes that would explicitly store the 'ts' for 'panelsID' and 'gesture' as values.

**Comment by daniel-2k** [comment]
I didn't understand what you are trying to do.
What do you do with the MQTT adapter? For what you pull the ts values via REST request?
What is the purpose of your request, what do you want to do?
Do you want to trigger an action depending of a panel gesture? The easiest way is to use a JavaScript trigger which can handel also triggers for the same panel and gesture because the state will be updated everytime the same event occur and you can raise a trigger for this. In in the JavaScript or Blockly you can do then everything you want.

**Comment by Radushka** [comment]
I am trying to trigger and automation (say increase a speaker volume or turn on a light) in Home Assistant when a panel is pressed. iobroker and the Nanoleaf adapter that you created are running in a docker instance. The only ways for Home Assistant to trigger events from a state in iobroker that I could figure out is either (1) request the values of the PanelID and Gesture on a regular interval via iobroker's REST adaptor or (2) have an MQTT broker adapter (in iobroker) that subscribes to the states of your adapter and publishes them for me to read them as MQTT sensors in Home assistant. I prefer the second option for speed and minimizing the load.

**Comment by daniel-2k** [comment]
I think there is a very detailed documentation of blockly in iobroker on GitHub:
https://github.com/ioBroker/ioBroker.javascript/blob/master/docs/en/blockly.md

**Comment by Radushka** [comment]
Yeah, that's my main source. Again, grateful for all your guidance. It opened a whole new world of opportunities for my smart home ;)
